www/apps/book/app/learn/fundamentals/modules/db-operations/page.mdx

```diff
@@ -683,3 +683,146 @@ class BlogModuleService {
+  status: "published",
+  metadata: {
+    featured: true
+  }
+})
+
+// Delete a post
+await postRepository.delete("post_123456")
+
+// Find multiple posts with filtering
+const posts = await postRepository.find({
+  where: { 
+    status: "published", 
+    author_id: "author_123"
+  },
+  take: 10,
+  skip: 0,
+  order: { created_at: "DESC" }
+})
+```
+
+You can also create custom repository classes with specialized query methods:
+
+```ts
+import { DALUtils } from "@medusajs/framework/utils"
+import { Post } from "./models/post"
+
+export class PostRepository extends DALUtils.mikroOrmBaseRepositoryFactory(Post) {
+  // Add custom methods for specialized queries
+  async findPublishedPosts() {
+    return await this.find({
+      where: { status: "published" }
+    })
+  }
+  
+  async findFeaturedPosts() {
+    return await this.find({
+      where: { 
+        metadata: {
+          featured: true
+        }
+      }
+    })
+  }
+  
+  async updateStatus(postId: string, status: string) {
+    return await this.update(postId, {
+      status
+    })
+  }
+}
+```
+
+### Complex Transaction Example
+
+Here's a more practical example demonstrating multiple operations in a single transaction:
+
+```ts
+await postRepository.transaction(async (transactionManager) => {
+  // Create a new blog post
+  const post = transactionManager.create(Post, {
+    title: "New Blog Post",
+    content: "This is the content of the post",
+    status: "draft"
+  })
+  
+  // Update author statistics in the same transaction
+  const authorStats = await transactionManager.findOne(AuthorStats, {
+    where: { author_id: post.author_id }
+  })
+  
+  authorStats.post_count = authorStats.post_count + 1
+  
+  // Save both changes atomically
+  await transactionManager.persistAndFlush([post, authorStats])
+  
+  return post
+})
+```
+
+### Advanced SQL Query Examples
+
+Here are examples of more complex SQL queries for a blog module:
+
+```ts
+// Get popular posts with their categories
+async function getPopularPosts() {
+  const knex = this.manager_.getKnex()
+  
+  return await knex.raw(`
+    SELECT 
+      p.id, 
+      p.title, 
+      p.created_at, 
+      c.name as category_name,
+      COUNT(cm.id) as comment_count
+    FROM posts p
+    LEFT JOIN post_categories pc ON p.id = pc.post_id
+    LEFT JOIN categories c ON pc.category_id = c.id
+    LEFT JOIN comments cm ON p.id = cm.post_id
+    WHERE p.status = 'published'
+    GROUP BY p.id, c.id
+    ORDER BY comment_count DESC
+    LIMIT 10
+  `)
+}
+
+// Calculate blog metrics
+async function getBlogMetrics() {
+  const knex = this.manager_.getKnex()
+  
+  return await knex.raw(`
+    SELECT 
+      DATE_TRUNC('month', p.created_at) as month,
+      COUNT(p.id) as posts,
+      COUNT(DISTINCT p.author_id) as unique_authors,
+      SUM(CASE WHEN p.status = 'published' THEN 1 ELSE 0 END) as published_posts
+    FROM posts p
+    GROUP BY month
+    ORDER BY month DESC
+    LIMIT 12
+  `)
+}
+```
```

Comment:
Use the entity manager as shown in the first section:

https://docs.medusajs.com/learn/fundamentals/modules/db-operations#run-queries

Also it would make sense to move this part into the Run Queries section.